compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9758,7 +9758,9 @@ private void handleSafeNavigation(BLangBlockStmt blockStmt, BLangAccessExpressio
             if (this.safeNavigationAssignment != null) {
                 this.safeNavigationAssignment.expr = addConversionExprIfRequired(accessExpr, tempResultVar.getBType());
             }
-            return;
+            if (Types.getReferredType(originalType).tag != TypeTags.JSON) {
```

Comment:
What about map<json>. There seems to be a check that has been added in L9753 which subsequently adds error to the access type, but I'm not sure if that's the best approach. We may have to rethink this block.